integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkJdbcIcebergIntegrationTest.java

```diff
@@ -0,0 +1,167 @@
+  }
+
+  @AfterAll
+  @SneakyThrows
+  public static void afterAll() {
+    SparkSession$.MODULE$.cleanupAnyExistingSession();
+    MockServerUtils.stopMockServer(mockServer);
+  }
+
+  @BeforeEach
+  @SneakyThrows
+  public void beforeEach() {
+    MockServerUtils.clearRequests(mockServer);
+    spark =
+        SparkSession.builder()
+            .master("local[*]")
+            .appName("IcebergIntegrationTest")
+            .config("spark.driver.host", LOCAL_IP)
+            .config("spark.driver.bindAddress", LOCAL_IP)
+            .config("spark.ui.enabled", false)
+            .config("spark.sql.shuffle.partitions", 1)
+            .config("spark.sql.warehouse.dir", "file:" + ICEBERG_PATH)
+            .config("spark.driver.extraJavaOptions", "-Dderby.system.home=" + ICEBERG_PATH)
+            .config("spark.openlineage.transport.type", "http")
+            .config(
+                "spark.openlineage.transport.url",
+                "http://localhost:" + mockServer.getPort() + "/api/v1/namespaces/iceberg-namespace")
+            .config("spark.openlineage.facets.debug.disabled", "false")
+            .config("spark.extraListeners", OpenLineageSparkListener.class.getName())
+            .config("spark.sql.defaultCatalog", "jdbc")
+            .config(
+                "spark.sql.extensions",
+                "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+            .config("spark.sql.catalog.jdbc.type", "jdbc")
+            .config("spark.sql.catalog.jdbc", "org.apache.iceberg.spark.SparkCatalog")
+            .config("spark.sql.catalog.jdbc.uri", "jdbc:sqlite:/tmp/iceberg_catalog.sqlite:")
+            .config("spark.sql.catalog.jdbc.warehouse", ICEBERG_PATH)
+            .config("spark.sql.sources.partitionOverwriteMode", "dynamic")
+            .config(
+                "spark.openlineage.dataset.removePath.pattern",
+                "(.*)(?<remove>\\_666)") // removes _666 from dataset name
+            .getOrCreate();
+  }
+
+  @Test
+  @DisabledIf("isJava8AndSpark35")
+  @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+  void testJdbcCatalog() {
+    createTempDataset(3).createOrReplaceTempView("temp");
+    spark.sql(
+        "CREATE TABLE IF NOT EXISTS jdbc.default.scan_source USING iceberg AS SELECT * FROM temp");
+    spark.sql(
+        "CREATE TABLE  IF NOT EXISTS jdbc.default.scan_target USING iceberg AS SELECT * FROM jdbc.default.scan_source");
+
+    List<OpenLineage.RunEvent> runEvents = getEventsEmittedWithJobName(mockServer, "scan_target");
+
+    assertThat(
+            runEvents.stream()
+                .flatMap(e -> e.getInputs().stream())
+                .filter(
+                    e ->
+                        hasSymlinkWith(
+                            e,
+                            symlink ->
+                                symlink.getName().endsWith("scan_source")
+                                    && symlink.getNamespace().startsWith("sqlite"))))
```

Comment:
Could we compare symlink to exact name & namespace?

Reply:
Updating this part right now actually.

Reply:
Done.

Reply:
No commits were pushed :smile: